docs/src/main/asciidoc/neo4j-guide.adoc

```diff
@@ -0,0 +1,545 @@
+If you have an already created project, the `neo4j` extension can be added to an existing Quarkus project with the `add-extension` command:
+
+[source,shell]
+----
+mvn quarkus:add-extension -Dextensions="neo4j"
+----
+
+Otherwise, you can manually add this to the dependencies section of your `pom.xml` file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-neo4j</artifactId>
+</dependency>
+----
+
+=== JSON Binding
+
+We will expose `Fruit` instances over HTTP in the JSON format.
+Consequently, you also need to add the `quarkus-resteasy-jsonb` extension:
+
+[source,shell]
+----
+mvn quarkus:add-extension -Dextensions="resteasy-jsonb"
+----
+
+== Configuring
+
+The Neo4j driver can be configured with standard Quarkus properties:
+
+[source]
+.src/main/resources/application.properties
+----
+quarkus.neo4j.uri = bolt://localhost:7687
+quarkus.neo4j.authentication.username = neo4j
+quarkus.neo4j.authentication.password = secret
+----
+
+You'll recognize the authentication here that you passed on to the docker command above.
+
+Having done that, the driver is ready to use.
+
+== Using the driver
+
+=== General remarks.
+
+The result of a statement consists usually of one or more `org.neo4j.driver.Record`.
+Those records contain arbitrary values, supported by the driver.
+If you return a node in the graph, it will be a `org.neo4j.driver.types.Node`.
+
+We add the following method to the `Fruit`, as a convenient way to create them:
+
+[source,java]
+----
+public static Fruit from(Node node) {
+    return new Fruit(node.id(), node.get("name").asString());
+}
+----
+
+Add a `FruitResource` skeleton like this and `@Inject` a `org.neo4j.driver.Driver` instance:
+
+[source,java]
+.src/main/java/org/acme/neo4j/FruitResource.java
+----
+package org.acme.neo4j;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.neo4j.driver.Driver;
+
+@Path("fruits")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class FruitResource {
+
+    @Inject
+    Driver driver;
+}
+----
+
+=== Reading nodes
+
+Add the following method to the fruit resource:
+
+[source,java]
+.src/main/java/org/acme/neo4j/FruitResource.java
+----
+@GET
+public CompletionStage<Response> get() {
+    AsyncSession session = driver.asyncSession(); // <1>
+    return session
+        .runAsync("MATCH (f:Fruit) RETURN f ORDER BY f.name")  // <2>
+        .thenCompose(cursor ->  // <3>
+            cursor.listAsync(record -> Fruit.from(record.get("f").asNode()))
+        )
+        .thenCompose(fruits ->  // <4>
+            session.closeAsync().thenApply(signal -> fruits)
+        )
+        .thenApply(Response::ok) // <5>
+        .thenApply(ResponseBuilder::build);
+}
+----
+<1> Open a new, asynchronous session with Neo4j
+<2> Execute a query. This is a Cypher statement.
```

Comment:
Maybe some link to the documentation on `Cypher`?

Reply:
Added to the top. Very good point, because I think Cypher is a great language, not only for our product. It will be available in Apache Spark and is an inspiration for the https://www.gqlstandards.org GQL effort